cmd/auth-handler.go

```diff
@@ -297,3 +332,40 @@ func (a authHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	writeErrorResponse(w, ErrSignatureVersionNotSupported, r.URL)
 }
+
+// isPutAllowed - check if PUT operation is allowed on the resource, this
+// call verifies bucket policies and IAM policies, supports multi user
+// checks etc.
+func isPutAllowed(atype authType, bucket, object string, r *http.Request) APIErrorCode {
+	switch atype {
+	case authTypeAnonymous:
+		if globalPolicySys.IsAllowed(policy.Args{
+			Action:          policy.PutObjectAction,
+			BucketName:      bucket,
+			ConditionValues: getConditionValues(r, ""),
+			IsOwner:         false,
+			ObjectName:      object,
+		}) {
+			return ErrNone
+		}
+	case authTypeSignedV2, authTypePresignedV2:
```

Comment:
this makes V2 signed requests bypass policy settings

Reply:
yep this is on purpose, STS is not supported for signature v2 in Amazon S3